### PR TITLE
Кудряшова Ирина. Задача 1. Вариант 9. Скалярное произведение векторов

### PR DESCRIPTION
**Pull request by Irina2004-tech**
Fixed seq_perf_tests
https://github.com/learning-process/ppc-2024-autumn/pull/69

### CODE REVIEW

**Review by allnes — APPROVED** [review]

**Review by aobolensk — APPROVED** [review]

File: tasks/seq/kudryashova_i_vector_dot_product/func_tests/vectorDotProductSeqFuncTests.cpp [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  taskDataSeq->inputs_count.emplace_back(vector1.size());
+  taskDataSeq->inputs_count.emplace_back(vector2.size());
+  taskDataSeq->outputs.emplace_back(reinterpret_cast<uint8_t *>(out.data()));
+  taskDataSeq->outputs_count.emplace_back(out.size());
+  // Create Task
+  kudryashova_i_vector_dot_product::TestTaskSequential testTaskSequential(taskDataSeq);
+  ASSERT_EQ(testTaskSequential.validation(), false);
+}
+
+TEST(kudryashova_i_vector_dot_product_seq, check_empty_vectors) {
```
Comment (ArinaTs) [inline tasks/seq/kudryashova_i_vector_dot_product/func_tests/vectorDotProductSeqFuncTests.cpp:106]: add a test for vectors of length 1 to seq

**Review by ArinaTs — APPROVED** [review]

File: tasks/mpi/kudryashova_i_vector_dot_product/func_tests/vectorDotProductMPIFuncTests.cpp [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    taskDataSeq->inputs_count.emplace_back(global_vector[1].size());
+    taskDataSeq->outputs.emplace_back(reinterpret_cast<uint8_t*>(reference.data()));
+    taskDataSeq->outputs_count.emplace_back(reference.size());
+    kudryashova_i_vector_dot_product_mpi::TestMPITaskSequential testMpiTaskSequential(taskDataSeq);
+    ASSERT_EQ(testMpiTaskSequential.validation(), true);
+    testMpiTaskSequential.pre_processing();
+    testMpiTaskSequential.run();
+    testMpiTaskSequential.post_processing();
+    ASSERT_EQ(kudryashova_i_vector_dot_product_mpi::vectorDotProduct(global_vector[0], global_vector[1]), result[0]);
+    ASSERT_EQ(reference[0], result[0]);
```
Comment (ascannel) [inline tasks/mpi/kudryashova_i_vector_dot_product/func_tests/vectorDotProductMPIFuncTests.cpp:63]: why do you need two identical checks?

File: tasks/mpi/kudryashova_i_vector_dot_product/func_tests/vectorDotProductMPIFuncTests.cpp [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
```
Comment (ascannel) [inline tasks/mpi/kudryashova_i_vector_dot_product/func_tests/vectorDotProductMPIFuncTests.cpp:278]: and also here

**Review by ascannel — CHANGES_REQUESTED** [review]

File: tasks/seq/kudryashova_i_vector_dot_product/func_tests/vectorDotProductSeqFuncTests.cpp (hunk 1 above)
Comment (Irina2004-tech) [reply tasks/seq/kudryashova_i_vector_dot_product/func_tests/vectorDotProductSeqFuncTests.cpp:129]: done

**Review by ascannel — APPROVED** [review]

File: tasks/seq/kudryashova_i_vector_dot_product/func_tests/vectorDotProductSeqFuncTests.cpp (hunk 1 above)
Comment (ArinaTs) [reply tasks/seq/kudryashova_i_vector_dot_product/func_tests/vectorDotProductSeqFuncTests.cpp:129]: Good job. I agree with that

**Pull request merged**